### ISSUE TITLE
Improve a11y documentation for button

### ISSUE THREAD

**Issue by kara** [angular/material2#245]
When using `md-button`, it's not obvious when you should use `<button md-button>` and when you should use `<a md-button>`.  We should better document how each affects accessibility.
